x-pack/plugins/security_solution/public/dashboards/pages/details/index.tsx

```diff
@@ -85,27 +86,30 @@ const DashboardViewComponent: React.FC = () => {
           data-test-subj="dashboard-view-wrapper"
         >
           <EuiFlexItem grow={false}>
-            <HeaderPage border title={dashboardDetails?.title ?? <EuiLoadingSpinner size="m" />}>
-              {showWriteControls && dashboardExists && (
-                <EditDashboardButton
-                  filters={filters}
-                  query={query}
-                  savedObjectId={savedObjectId}
-                  timeRange={timeRange}
-                />
-              )}
-            </HeaderPage>
+            {dashboardContainer && (
+              <DashboardContainerContextProvider dashboardContainer={dashboardContainer}>
+                <>
+                  <HeaderPage
+                    border
+                    title={<DashboardTitle onTitleLoaded={setDashboardTitle} />}
+                    subtitle={<DashboardToolBar onLoad={onDashboardToolBarLoad} />}
+                  />
+                </>
+              </DashboardContainerContextProvider>
```

Comment:
This context looks a bit an overkill. Can't we just pass the `dashboardContainer` to `DashboardTitle`and `DashboardToolBar` by props here?